freqtrade/commands/cli_options.py

```diff
@@ -548,4 +548,20 @@ def __init__(self, *args, **kwargs):
         help='Do not print epoch details header.',
         action='store_true',
     ),
+    # TODO-mg: Set up these options
+    # "leverage": Arg(
+    #     '-L', '--leverage',
+    #     help='Leverage applied for margin and futures trading.',
+    #     action=''
+    # ),
+    # "leverage_min": Arg(
+    #     '--leverage-min',
+    #     help='The minimum leverage to trade at if the desired leverage is not available',
+    #     action=''
+    # ),
+    # "shorts_enabled": Arg(
+    #     '--shorts-enabled',
+    #     help='Enables short trading',
+    #     action=''
+    # )
```

Comment:
please don't add these as cli arguments (at least not for now).

Leverage (+ short support) should come from the strategy - as it's a strategy decision IF it works with shorts or not - not a configuration / cli option.

Also - for leverage itself - that's not a configuration option either - as different pairs allow different levels of leverage - so it'll probably be a strategy-level callback - with the possible leverage values provided (look at kraken - some pairs have 2,3,4,5 - others only 2,3 , ....

Reply:
But the user has to decide how much leverage they want to use. I don't think it should be either trade with the max available leverage or don't trade with leverage at all and it would be wrong to assume that the user wants to trade at 5x leverage if 5x leverage is available. 

I think the best way to go about it would be to have the user provide their desired leverage in the configuration as a range
*  there would be config.min_leverage
* and config.leverage (or config.max_leverage)

If the maximum available leverage for a pair is lower than config.min_leverage:
    then bot won't trade on that pair
    
If the maximum available leverage for a pair is ≥ config.min_leverage:
    then the bot with trade the pair at the maximum allowable leverage which is ≤ config.max_leverage

Reply:
i'm not saying "use what's available" - but implement it as a callback (so the strategy get's "asked" what leverage to use) - similar to `custom_sell` (or so).

Having this in the configuration will make it static (applies to all pairs) - which will be highly limiting and inflexible ( think - use  3x leverage for ETH - but 5x on XRP because the indicators there show better chances).

Reply:
@xmatthias That seems a lot more complicated than what I was going for. Is there a way to do both? My plan was to write it just so that it would trade like normal, but apply the maximum desired leverage on every trade. I feel like there's a use case for both. I think I want leave the option open to define a static leverage and the user could also use the strategy implemented leverage if they want

What if leverage was in the config, and then there was an option like "strategy_implemented" or something?

Reply:
Oh wait, the strategy could just always return a static value then

Reply:
That's my point :) 
the problem with "static" is that it'll not work for all pairs (at least not on non-binance exchange) - as (max) leverage depends on the pair.